cloudinit/cmd/main.py

```diff
@@ -334,6 +337,90 @@ def _should_bring_up_interfaces(init, args):
     return not args.local
 
 
+def _should_wait_via_cloud_config(
+    raw_config: Optional[Union[str, bytes]]
+) -> Tuple[bool, Reason]:
+    if not raw_config:
+        return False, "no configuration found"
+
+    # If our header is anything other than #cloud-config, wait
+    possible_header: Union[bytes, str] = raw_config.strip()[:13]
+    if isinstance(possible_header, str):
+        decoded_header = possible_header
+    elif isinstance(possible_header, bytes):
+        try:
+            decoded_header = possible_header.decode("utf-8")
+        except UnicodeDecodeError:
+            return True, "Binary user data found"
+    if not decoded_header.startswith("#cloud-config"):
+        return True, "non-cloud-config user data found"
```

Comment:
This reminds me of [type_from_starts_with](https://github.com/canonical/cloud-init/blob/main/cloudinit/handlers/__init__.py#L291). But, your approach is more specific and a teeny bit more efficient since you are only looking at the first 13 characters and only checking for `#cloud-config`. 

If you wanted to alter this the common utility:
```suggestion
     try:
         possible_header = util.decode_binary(raw_config.strip()[:13])
     except UnicodeDecodeError:
         return True, "Binary user data found"
      if "text/cloud-config" != type_from_starts_with(possible_header):
         return True, "non-cloud-config user data found"
```

Reply:
Applied the change with the try as currently the binary distinction doesn't actually matter. Know that it's non-cloud-config is good enough.